ci/infra/libvirt/master-instance.tf

```diff
@@ -101,29 +101,41 @@ resource "libvirt_domain" "master" {
   }
 }
 
-resource "null_resource" "master_wait_cloudinit" {
+resource "null_resource" "master" {
   depends_on = [libvirt_domain.master]
   count      = var.masters
 
+  triggers = {
+    master_ips = "${join(",", libvirt_domain.master.*.network_interface.0.addresses.0)}"
+    username   = var.username
+  }
+
   connection {
     host = element(
-      libvirt_domain.master.*.network_interface.0.addresses.0,
-      count.index
+      split(",", self.triggers.master_ips),
+      count.index,
     )
-    user     = var.username
-    password = var.password
-    type     = "ssh"
+    user  = self.triggers.username
```

Comment:
Why use here triggers instead of the var?

Reply:
This avoids using external references (variables, resource attributes...) with the destroy provisioners which is deprecated.